### ISSUE TITLE
decorana yields different outputs for different versions of vegan

### ISSUE THREAD

**Issue by jannes-m** [vegandevs/vegan#303]
Dear Jari,
first of all, thanks for your fantastic package which I have probably been using for more than ten years. 
I have stumbled across some inconsistency regarding the output of `decorona`. Specifically, `decorana` returns different results since **vegan** package version 2.5-1 compared to package version 2.4-6.

So, first of all, let's download and install **vegan** 2.4-6 and subject the `varechem` dataset to a DCA. I save the `decorana` output in a directory created by **rappdirs**.

``` r
devtools::install_version("vegan", version = "2.4-6",
                          repos = "http://cran.us.r-project.org")
#> Downloading package from url: http://cran.us.r-project.org/src/contrib/Archive/vegan/vegan_2.4-6.tar.gz
#> Installing package into '/home/jannes/R/x86_64-pc-linux-gnu-library/3.5'
#> (as 'lib' is unspecified)
library("vegan")
#> Loading required package: permute
#> Loading required package: lattice
#> This is vegan 2.4-6
packageVersion("vegan")
#> [1] '2.4.6'
# attach varespec data
data("varespec")
# compute DCA
dca_1 = decorana(varespec, iweigh = 1)
# save it to a "temporary" file
saveRDS(dca_1, file.path(rappdirs::user_data_dir(), "dca_1.rds"))
```

<sup>Created on 2019-01-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Secondly, I have terminated R and RStudio and opened RStudio again (pls note that a simple R session restart is not sufficient, instead one has to properly close R and RStudio). This time, I will install **vegan** from CRAN, run again a DCA on the `varespec` dataset and compare the results between **vegan** package versions 2.4-6 and 2.5-3.

``` r
devtools::install_cran("vegan")
#> Installing 1 packages: vegan
#> Installing package into '/home/jannes/R/x86_64-pc-linux-gnu-library/3.5'
#> (as 'lib' is unspecified)
library("vegan")  # 2.5.3
#> Loading required package: permute
#> Loading required package: lattice
#> This is vegan 2.5-3
packageVersion("vegan")
#> [1] '2.5.3'

# attach varespec and run the same DCA again
data("varespec")
dca_2 = decorana(varespec, iweigh = 1)

# attach first dca result
dca_1 = readRDS(file.path(rappdirs::user_data_dir(), "dca_1.rds"))

# compare DCA results 
all.equal(scores(dca_1), scores(dca_2))
#> [1] "Mean relative difference: 1.786744"
par(mfrow = c(1, 2))
plot(dca_1)
plot(dca_2)
```

![](https://i.imgur.com/VX5hShJ.png)

<sup>Created on 2019-01-16 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

It turns out that the results differ substantially. So the question is which result is correct? Looking at the NEWS, is it possible that a bug was introduced when increasing the performance of `decorana` in package version 2.5-1?

**Comment by jarioksa** [comment]
Yes, this is a bug, and it concerns downweighted analysis (argument `iweigh = 1`): the row and column sums must be evaluated after downweighting. The bug is fixed in commit 0de6bc48f394416cd085130c4873a86d33d2f376 and the fix will be ported to vegan_2.5-4.

**Comment by jannes-m** [comment]
Thanks for the quick answer and the quick fix!!
